examples/schema_template_anp_epg/main.tf

```diff
@@ -13,25 +13,52 @@ provider "mso" {
   insecure = true
 }
 
+data "mso_tenant" "common" {
+  name         = "common"
+  display_name = "common"
+}
+
+resource "mso_schema" "schema1" {
+  name          = "tf_test_schema"
+  template_name = "Template1"
+  tenant_id     = data.mso_tenant.common.id
+}
+
+resource "mso_schema_template_anp" "anp1" {
+  schema_id    = mso_schema.schema1.id
+  template     = mso_schema.schema1.template_name
+  name         = "anp1"
+  display_name = "anp1"
+}
+
 resource "mso_schema_template_anp_epg" "anp_epg" {
-  schema_id                  = "5eafca7d2c000052860a2902"
-  template_name              = "stemplate1"
-  anp_name                   = "sanp1"
-  name                       = "nkusegte"
-  bd_name                    = "testBD"
-  vrf_name                   = "vrf1"
-  display_name               = "nkuseg"
-  useg_epg                   = true
-  intra_epg                  = "enforced"
-  intersite_multicast_source = true
-  proxy_arp                  = true
-  preferred_group            = true
-  bd_template_name           = "stemplate1"
-  vrf_schema_id              = "5eafeb792c0000a18e0a2900"
-  bd_schema_id               = "5eafeb792c0000a18e0a2900"
-  vrf_template_name          = "stemplate1"
-  access_type= "private"
-  deployment_type = "cloud_native"
-  service_type = "custom"
-  custom_service_type = "Test"
+  schema_id     = mso_schema.schema1.id
+  template_name = mso_schema.schema1.template_name
+  anp_name      = mso_schema_template_anp.anp1.name
+  name          = "epg1"
+  display_name  = "epg1"
+  bd_name       = mso_schema_template_bd.bd1.name
+  # bd_template_name           = mso_schema_template_bd.bd1.schema_id
+  # bd_schema_id               = mso_schema_template_bd.bd1.template_name
+  vrf_name = mso_schema_template_vrf.vrf1.name
+  # vrf_schema_id              = mso_schema_template_vrf.vrf1.schema_id
+  # vrf_template_name          = mso_schema_template_vrf.vrf1.template
+}
+
+resource "mso_schema_template_bd" "bd1" {
```

Comment:
mso_schema_template_bd and mso_schema_template_vrf should go before mso_schema_template_anp_epg also we can remove the commented lines if not necessary.

Reply:
The order does not matter in Terraform but for ease of reading I have made the change.